dotenv/main.py

```diff
@@ -181,21 +181,42 @@ def unset_key(dotenv_path, key_to_unset, quote_mode="always"):
 
 
 def resolve_nested_variables(values):
-    def _replacement(name):
+    def _replacement(name, default=None):
         """
         get appropriate value for a variable name.
         first search in environ, if not found,
         then look into the dotenv variables
         """
-        ret = os.getenv(name, new_values.get(name, ""))
-        return ret
+
+        # If name is set as an environment variable, use its value
+        if name in os.environ:
+            return os.getenv(name)
+
+        # If name is the same as k, we need to handle the situation special, or we will have ${K}
+        # or ${K:-default} as the return value.  Note that k comes from the for loop below
+        if name == k:
```

Comment:
`k` isn't defined

Reply:
It is; it comes from the for loop that iterates over `values` right after the definition of `_re_sub_callback()`. Also see the end of the comment above this line.

That’s exactly the reason i wrote it feels hacky.